pdr_backend/analytics/predictoor_stats.py

```diff
@@ -158,46 +158,48 @@ def get_endpoint_statistics(
 
     return overall_accuracy, pair_timeframe_stats, predictoor_stats
 
-
 @enforce_types
-def get_cli_statistics(all_predictions: List[Prediction]) -> None:
-    total_predictions = len(all_predictions)
+def get_feed_summary_stats(predictions_df: pl.DataFrame) -> pl.DataFrame:
+    # 1 - filter from lake only the rows that you're looking for
+  df = predictions_df.filter(~((pl.col("trueval").is_null()) | (pl.col("payout").is_null())))
 
-    stats, correct_predictions = aggregate_prediction_statistics(all_predictions)
+  print(df)
 
-    if total_predictions == 0:
-        print("No predictions found.")
-        return
+  # Group by pair
+  df = df.groupby("pair").agg(
+    pl.col("timeframe").first().alias("timeframe"),
+    pl.col("source").first().alias("source"),
+    pl.col("payout").sum().alias("sum_payout"),
+    pl.col("stake").sum().alias("sum_stake"),
+    pl.col("prediction").count().alias("num_predictions"),
+    (pl.col("prediction").sum() / pl.col("pair").count() * 100).alias("accuracy")
+  )
 
-    if correct_predictions == 0:
-        print("No correct predictions found.")
-        return
+  print(df)
+  return df
 
-    print(f"Overall Accuracy: {correct_predictions/total_predictions*100:.2f}%")
 
-    for key, stat_pair_timeframe_item in stats["pair_timeframe"].items():
-        pair, timeframe = key
-        accuracy = (
-            stat_pair_timeframe_item["correct"]
-            / stat_pair_timeframe_item["total"]
-            * 100
-        )
-        print(f"Accuracy for Pair: {pair}, Timeframe: {timeframe}: {accuracy:.2f}%")
-        print(f"Total stake: {stat_pair_timeframe_item['stake']}")
-        print(f"Total payout: {stat_pair_timeframe_item['payout']}")
-        print(f"Number of predictions: {stat_pair_timeframe_item['total']}\n")
-
-    for predictoor_addr, stat_predictoor_item in stats["predictor"].items():
-        accuracy = stat_predictoor_item["correct"] / stat_predictoor_item["total"] * 100
-        print(f"Accuracy for Predictoor Address: {predictoor_addr}: {accuracy:.2f}%")
-        print(f"Stake: {stat_predictoor_item['stake']}")
-        print(f"Payout: {stat_predictoor_item['payout']}")
-        print(f"Number of predictions: {stat_predictoor_item['total']}")
-        print("Details of Predictions:")
-        for detail in stat_predictoor_item["details"]:
-            print(f"Pair: {detail[0]}, Timeframe: {detail[1]}, Source: {detail[2]}")
-        print("\n")
 
+@enforce_types
+def get_predictoor_summary_stats(predictions_df: pl.DataFrame) -> pl.DataFrame:
+  # 1 - filter from lake only the rows that you're looking for
+  df = predictions_df.filter(~((pl.col("trueval").is_null()) | (pl.col("payout").is_null())))
+
+  print(df)
+
+  # Group by pair
+  df = df.groupby("pair").agg(
```

Comment:
Have you checked/verified the print/output?
Perhaps we can verify this in a test... but...

I believe you'll have to group by ["pair", "timeframe"] such as:
```
df = df.groupby(["pair", "timeframe"]).agg(
```